user_logic.py

```diff
@@ -0,0 +1,36 @@
+import db
+
+
+def valid_user(user_id):
+    """Check whether the user is in the tabe and return the name."""
+    SQL = """SELECT user_name FROM users WHERE id = %s;"""
+    data = (user_id,)
+    fetch = "cell"
+    user_name = db.run_statements(((SQL, data, fetch),))[0]
```

Comment:
This line might throw DatabaseError or TypeError.
Examples:
user_id = "just_a_string" --> DatabaseError
user_id = 666 --> db.run_statments will try to subscript a None object since id = 666 does not exist in users table --> TypeError

Example code for question_id validation:

def valid_question_id(question_id):
    """Return True if question_id found in question table."""
    SQL = """SELECT id FROM question WHERE id = %s;"""
    data = (question_id,)
    fetch = "one"
    try:
        found_id = db.run_statements(((SQL, data, fetch),))[0][0]
    except (DatabaseError, TypeError):
        return False
    return True